### ISSUE TITLE
Support nested DeltaDeletedResource

### ISSUE THREAD

**Issue by abgreg** [OData/AspNetCoreOData#812]
I have an entity assignmentSettings which has a property gradingCategories that is a list of sub-entities.

This works great:
PATCH /assignmentSettings/gradingCategories
```json
{
    {
    "@odata.context": "{{api}}/v1.0/edu/$metadata#gradingCategories/$delta",
    "value": [
        {
            "@odata.context":"{{api}}/v1.0/edu/$metadata#gradingCategories/$deletedEntity",
            "id": "{{gradingCategoryId0}}",
            "reason": "deleted"
        }
    ]
}
```

But if I try it at the parent level, the ModelState reports error: "System.ArgumentException: The value "Microsoft.AspNetCore.OData.Deltas.DeltaDeletedResource`1[Microsoft.Edu.Assignments.Application.Contracts.AssignmentSettings.GradingCategoryModel]" is not of type "Microsoft.Edu.Assignments.Application.Contracts.AssignmentSettings.GradingCategoryModel"
 and cannot be used in this generic collection. (Parameter 'value')"
PATCH /assignmentSettings
```json
{
    "gradingCategories@delta": [
        {
            "@odata.context":"{{api}}/v1.0/edu/$metadata#gradingCategories/$deletedEntity",
            "id": "{{gradingCategoryId0}}",
            "reason": "deleted"
        }
    ]
}
```

**Comment by julealgon** [comment]
...I've never heard of this "x@delta" notation before nor the `DeltaDeletedResource` 😮 

Is this documented anywhere in MSDN or in a blog post? I'd love to see how this works.

Sorry for the off-topic comment.

**Comment by abgreg** [comment]
@julealgon The idea is to allow adding, updating, and deleting multiple entities in a single request.
Here's a bit of information @xuzhg shared when I was first working on this.
Sample using OData's 4.0 notation: https://github.com/OData/AspNetCoreOData/blob/main/sample/ODataRoutingSample/Controllers/v1/OrganizationsController.cs#L94
And the 4.1 request body format by comparison:
http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_DeletedEntity

**Comment by xuzhg** [comment]
@abgreg The current 8.0.12 should work with 'EnableReadingODataAnnotationWithoutPrefix' setting.

I have PR that enables EnableReadingODataAnnotationWithoutPrefix by default.

@julealgon In the PR, I share an example of the nested delta. Please let me know any questions.

**Comment by julealgon** [comment]
> @julealgon In the PR, I share an example of the nested delta. Please let me know any questions.

@xuzhg Sam do you think a blog post could be made about this feature? It looks like such a massively useful (and complex) one. I'm also curious if it fully replaces `JsonPatch`. Would be awesome to have a walkthrough of it in detail along with documented examples and could also serve as good press for the framework.

**Comment by xuzhg** [comment]
@julealgon That's a good suggestion.  Actually, OData team has a post at: https://devblogs.microsoft.com/odata/bulk-operations-support-in-odata-web-api/

Of course, That's for 7.x version. In the 8, I haven't ported the Data handler because I haven't a different opinion for handler.

When it's ready, I will create a post based on 8 version.

**Comment by abgreg** [comment]
@xuzhg helped me get to a fix: Upgrade from OData 8.0.10 -> 8.0.12. :)

**Comment by gathogojr** [comment]
Fixed in 8,0.12
